packages/app-contracts/src/Codes/Add.tsx

```diff
@@ -0,0 +1,118 @@
+// Copyright 2017-2019 @polkadot/app-contracts authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import React from 'react';
+import { Button, Input } from '@polkadot/ui-app';
+import { Hash } from '@polkadot/types';
+
+import ContractModal, { ContractModalProps, ContractModalState } from '../Modal';
+
+import ValidateCode from './ValidateCode';
+import store from '../store';
+import translate from '../translate';
+
+type Props = ContractModalProps;
+
+type State = ContractModalState & {
+  codeHash: string,
+  isBusy: boolean,
+  isCodeValid: boolean
+};
+
+class Add extends ContractModal<Props, State> {
+  constructor (props: Props) {
+    super(props);
+    this.defaultState = {
+      ...this.defaultState,
+      codeHash: '',
+      isBusy: false,
+      isCodeValid: false
+    };
+    this.state = this.defaultState;
+  }
+
+  headerText = 'Add an existing code hash';
```

Comment:
Move to where the `state` & `defaultState` is defined, i.e. right at the start of the component, first.

Reply:
This one is a bit problematic, would suggest setting it in the constructor so it is translated.